firmware/dfu.ksy

```diff
@@ -0,0 +1,77 @@
+meta:
+  id: dfu
+  title: Device Firmware Upgrade
+  file-extension: dfu
+  license: CC0-1.0
+  endian: le
+doc: |
+  Device Firmware Upgrade files
+
+  Test files: <https://micropython.org/download/espruino_pico/>
+doc-ref: https://raw.githubusercontent.com/micropython/micropython/247d7e2/tools/pydfu.py
+seq:
+  - id: magic
+    size: 5
+    contents: "DfuSe"
+  - id: version
+    type: u1
+    valid: 1
+  - id: len_dfu
+    -orig-id: size
+    type: u4
+  - id: num_targets
+    type: u1
+  - id: targets
+    type: target
+    repeat: expr
+    repeat-expr: num_targets
+  - id: device
+    doc: Firmware version
+    type: u2
+  - id: product
+    type: u2
+  - id: vendor
+    type: u2
```

Comment:
should it be an enum?

Reply:
This refers to USB ids, there are quite a few of those (many thousands) and I am not so sure that would help here.

Reply:
We should generate a file with them automatically. I don't remember if I have written a script for that, if not, probably I should.